qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -1157,18 +1161,35 @@ def least_busy(
         Raises:
             QiskitBackendNotFoundError: If no backend matches the criteria.
         """
-        backends = self.backends(
-            min_num_qubits=min_num_qubits, instance=instance, filters=filters, **kwargs
-        )
+        if not self._backends_list:
+            self._backends_list = self._active_api_client.list_backends()
+
         candidates = []
-        for back in backends:
-            backend_status = back.status()
-            if not backend_status.operational or backend_status.status_msg != "active":
-                continue
-            candidates.append(back)
+        for backend in self._backends_list:
+            if backend["status"]["name"] == "online" and backend["status"]["reason"] == "available":
+                candidates.append(backend)
+        if filters:
+            # filters will still be slow because we need the backend configs
+            backends = self.backends(
+                min_num_qubits=min_num_qubits, filters=filters, instance=instance, **kwargs
+            )
+            filtered_backend_names = [back.name for back in backends]
+            for candidate in candidates.copy():
+                if candidate["name"] not in filtered_backend_names:
+                    candidates.remove(candidate)
```

Comment:
I need to look into the filters, but is there any additional filter that we can apply directly to `self._backends_list` without having to get the configs?

Reply:
Here's what the new backends response returns - so I think we can also apply `is_simulator` and `processor_type`? Those are both in the backend config. 

```
{
      "name": "ibmq_nazca",
      "status": {
        "name": "paused",
        "reason": "Maintenance"
      },
      "is_simulator": false,
      "qubits": 16,
      "clops": {
        "type": "hardware",
        "value": 80
      },
      "processor_type": {
        "family": "Quantum",
        "revision": "RevB",
        "segment": "Segment2"
      },
      "queue_length": 20
    }
```

Reply:
Let's do it in a follow up